### DISCUSSION TITLE
New BPMN Pool formatting questions.

### DISCUSSION THREAD

**Discussion by gordon-brown** [clientIO/joint#2413] [Q&A]
### Introduction

I am writing a program to convert Syncfusion Diagrams to JointJS diagrams.  This is a server side application using Newton JsonSoft to convert the JSON from one format to the other. Currently I am working on converting a simple swimlane. I have taken the demo provided by Roman and added functionality to import and export joint JS file. I have provided screenshots of our syncfusion diagram and the jointjs diagram generated by our conversion. I am unsure how to set formatting properties for the pool, lanes and phases. The syncfusion diagram has some fill color properties set. I tried setting the fill color property in the JointJS diagram. I will need to do the same with font, fontsize, bold, italics and possibly otheres. Attached is the joint file I generated. 

![JointJSSwimlane](https://github.com/clientIO/joint/assets/32515524/6460dea5-8310-434d-81c8-41e15efa2b14)

![SyncfusionSwimlane](https://github.com/clientIO/joint/assets/32515524/a37ea19f-6c5f-4362-a7e6-4308c03e1aea)

[Swimlane.joint.txt](https://github.com/clientIO/joint/files/13548961/Swimlane.joint.txt)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
In the BPMN Pools demo each:
- **Swimlane** is a `bpmn2.Pool`. ([docs](https://resources.jointjs.com/docs/rappid/v3.7/shapes.bpmn2.html#HeaderedPool))
- **Phase** is a `standard.Rectangle` ([docs](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#shapes.standard.HeaderedRectangle))
- **Pool** is also a `standard.Rectangle`

Here is an example of how to change the colors on them:
```js
swimalne1.attr({
  header: { fill: 'red' },
  lane_0: { fill: 'red' },
});

phase1.attr({
   header: { fill: 'green' }
});

pool1.attr({
   header: { fill: 'blue' }
});
```

**Reply by gordon-brown** [reply]
Thanks Roman. I was thinking these were new shapes.
